template/logEntry/logEntry.proto

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package logEntry;
+
+// TODO: STUFF THAT IS COMMENTED OUT NEEDS TO BE TURNED ON ONCE CODE GEN IS FIXED
```

Comment:
Instead, I think the template needs to look like this:

```proto
message Template {
    string severity = 2; // expression-populated value that must have ValueType `STRING`
    map<string, istio.mixer.v1.config.descriptor.ValueType> labels = 3;
}
```
(and the generated Instance has a `name` that we use as the log name field)

Then a logging adapter's configuration might look like:

```proto
message Params {
    map<string, my.backend.Severity> severity_map = 1; // maps string expression result to a severity value for this backend
    map<string, my.backend.Format> format_map = 2; // maps from instance name to format for that instance
    map<string, string> log_templates = 3; // maps from instance name to log template for that instance
}
```

We don't need to handle timestamps specifically as they can just be dimensions and derived from attributes (which is what you'd want to do for the log entry anyway; use `request.time` for start time and `request.duration` to compute the end time).